src/utils/aria.ts

```diff
@@ -0,0 +1,45 @@
+const ifAriaNoLabel = (ariaLabel?: string) => !!ariaLabel || ariaLabel === '';
+
+export const ariaLabelOrHidden = (ariaLabel?: string) => {
+  return ifAriaNoLabel(ariaLabel)
+    ? { 'aria-label': ariaLabel, role: 'img' }
+    : { 'aria-hidden': true };
+};
+
+/**
+ * Set element ability to be focusable based on aria-label
+ * @param {String} ariaLabel optional aria-label
+ */
+export const ariaFocusableNoLabel = (ariaLabel?: string) => {
+  return ifAriaNoLabel(ariaLabel) ? {} : { focusable: false };
+};
+
+/**
+ * Returns object with 'aria-describedby' which can be spread to props.
+ * E.g:
+ * @example
+ * <Component
+ *   {...getAriaDescribedBy([
+ *      describingElement ? "id-of-describing-element" : undefined,
+ *      otherDescribingElement ? "id-of-other-describing-element" : undefined,
+ *   ])}
+ * />
+ * @param describedByIds Array of id-strings
+ * @returns Object with 'aria-describedby' if there is atleast one value that is not undefined. Otherwise returns empty Object.
+ */
+export const getAriaDescribedByProp = (
+  describedByIds: (string | undefined)[],
+):
+  | { 'aria-describedby': string }
+  | {
+      'aria-describedby'?: undefined;
```

Comment:
should this return type rather be just `{}` if that is not allowed by linters, isn't the latter option enough, since it covers both cases: an empty object and aria-describedby string?

Reply:
If using the latter, it will complain:
`Type 'string' is not assignable to type 'undefined'.ts(2322)`

If returning type is `{}` it does work, but then JSDoc obviously can not show the return type that it will have that `aria-describedby` inside of it if that is the case. But this tool is for internal use, so we can live without this information too.

Reply:
Yeah, was just interested why the typing was done this way.